packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -621,6 +621,13 @@ export class CronjobController extends BaseController<
       // immediately.
       if (event.recurring && eventDate <= now) {
         this.#execute(event);
+        return;
+      }
+
+      // If the existing event has not passed, start the timer.
+      if (eventDate >= now) {
+        this.#startTimer(event);
```

Comment:
Without this we would ALWAYS reschedule cronjobs, even if their current date was valid